find_includes.rb

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env ruby
+
+# @ivoanjo: I've used ChatGPT to generate this quick-and-dirty tool. It can be used to identify recursively which
+# ruby headers include which headers, and which are not used. E.g. by using
+# `find_includes.rb vm_core.h iseq.h ractor_core.h` we get a rough list of headers that are not needed by the
+# datadog gem. Always remember to validate the result -- this tool isn't perfect (for instance it doesn't detect that
+# thread_pthread.h is in use).
```

Comment:
> removing files that are not actually needed by the dd-trace-rb gem to work

This seems a bit backwards: `datadog-ruby_core_source` is a dependency of `dd-trace-rb`, but this introduces a circular dependency back from `datadog-ruby_core_source` to `dd-trace-rb`.

The mention that the tool isn't perfect† doesn't bode well either. It is also AIUI non-reversible, not telling you that suddenly you need a header that isn't there.

Combined with the huge resulting diff, I am not confident with a merge.

† Given that C99 preprocessing is sort of Turing complete the only way to surefire know is by running a full-blown C preprocessor, like `cc -E`...